server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -268,6 +288,51 @@ class InteractiveSession(
     }
   }
 
+  private def datanucleusJars(livyConf: LivyConf): ArrayBuffer[String] = {
+    if (sys.env.getOrElse("LIVY_INTEGRATION_TEST", "false").toBoolean) {
+      // datanucleus jars has already been in classpath in integration test
+      ArrayBuffer.empty
+    } else {
+      val jars = ArrayBuffer.empty[String]
+      val sparkHome = livyConf.sparkHome().get
+      val libdir =
+        if (new File(sparkHome, "RELEASE").isFile) {
+          new File(sparkHome, "lib")
+        } else {
+          new File(sparkHome, "lib_managed/jars")
+        }
+
+      if (libdir.isDirectory()) {
+        jars ++= libdir.listFiles().filter(_.getName.startsWith("datanucleus-"))
+          .map(_.getAbsolutePath)
+      }
+      require(jars.nonEmpty, "datanucleus jars can not be found")
```

Comment:
This will break CDH since it doesn't ship datanucleus jars in Spark's lib directory. This shouldn't be a requirement anyway since most people use proper metastore servers (or at least most people I know).